src/Mobs/Path.cpp

```diff
@@ -0,0 +1,391 @@
+// TODO uniq_ptr instead of ptr whenever possible
+
+#ifndef COMPILING_PATHFIND_DEBUGGER
+/* MCServer headers */
+#include "Globals.h"
+#include "../World.h"
+#include "../Chunk.h"
+#endif
+
+
+#include "Path.h"
+
+#include <stdio.h>
+#include <exception>
+#include <cmath>
+
+#define DISTANCE_MANHATTEN 0  // 1: More speed, a bit less accuracy 0: Max accuracy, less speed.
+#define HEURISTICS_ONLY 0  // 1: Much more speed, much less accurate.
+#define CALCULATIONS_PER_STEP 5  // Higher means more CPU load but faster path calculations.
+// The only version which guarantees the shortest path is 0, 0.
+
+enum class eCellStatus {OPENLIST,  CLOSEDLIST,  NOLIST};
+struct cPathCell
+{
+	Vector3d m_Location;   // Location of the cell in the world
```

Comment:
Correct, int coords only right now. I used Vector3d because non ints are in the roadmap. But now that I think of it, some parts of the code will always be integral either way.